macro/run_simScript.py

```diff
@@ -650,6 +650,41 @@
  rc1 = os.system("rm  "+outFile)
  rc2 = os.system("mv "+tmpFile+" "+outFile)
  fin.SetWritable(False) # bpyass flush error
+
+if simEngine == "muonDIS":
+    f_outputfile = ROOT.TFile.Open(outFile, "read")
+    output_tree = f_outputfile.cbmsim
+
+    f_muonfile = ROOT.TFile.Open(inputFile, "read")
+    muondis_tree = f_muonfile.Get("DIS")
+
+    iMuon = ROOT.TClonesArray("TVectorD")
+    muondis_tree.SetBranchAddress("InMuon", iMuon)
+
+    temp_filename = outFile.replace(".root", ".tmp")
+    f_temp = ROOT.TFile.Open(temp_filename, "recreate")
+
+    new_tree = output_tree.CloneTree(0)
+
+    cross_section = array("f", [0.0])
+    cross_section_leaf = new_tree.Branch(
+        "CrossSection", cross_section, "CrossSection/F"
+    )
+
+    for output_event, muondis_event in zip(output_tree, muondis_tree):
+        mu = muondis_event.InMuon[0]
+        cross_section[0] = mu[10]
+        new_tree.Fill()
+
+    new_tree.Write("", ROOT.TObject.kOverwrite)
+    f_temp.Close()
+    f_outputfile.Close()
+    f_muonfile.Close()
+
+    os.replace(temp_filename, outFile)
```

Comment:
It might be good to use context managers here to ensure all files are left in a good state even if something fails.

For the temporary file, maybe https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile would be good to use, but we can probably also handle the cleanup ourselves using a context manager and `finally`.